precompiles/erc20/integration_test.go

```diff
@@ -0,0 +1,1048 @@
+package erc20_test
+
+import (
+	"fmt"
+	"math/big"
+
+	auth "github.com/evmos/evmos/v15/precompiles/authorization"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v15/precompiles/erc20"
+	"github.com/evmos/evmos/v15/precompiles/erc20/testdata"
+	"github.com/evmos/evmos/v15/precompiles/testutil"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/factory"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/keyring"
+	utiltx "github.com/evmos/evmos/v15/testutil/tx"
+	evmtypes "github.com/evmos/evmos/v15/x/evm/types"
+
+	//nolint:revive // dot imports are fine for Ginkgo
+	. "github.com/onsi/ginkgo/v2"
+	//nolint:revive // dot imports are fine for Ginkgo
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ERC20 Extension -", func() {
+	var (
+		contractAddr common.Address
+		err          error
+		sender       keyring.Key
+
+		execRevertedCheck testutil.LogCheckArgs
+		failCheck         testutil.LogCheckArgs
+		passCheck         testutil.LogCheckArgs
+	)
+
+	BeforeEach(func() {
+		s.SetupTest()
+
+		sender = s.keyring.GetKey(0)
+
+		contractAddr, err = s.factory.DeployContract(
+			sender.Priv,
+			evmtypes.EvmTxArgs{}, // NOTE: passing empty struct to use default values
+			factory.ContractDeploymentData{
+				Contract:        testdata.ERC20AllowanceCallerContract,
+				ConstructorArgs: []interface{}{s.precompile.Address()},
+			},
+		)
+		Expect(err).ToNot(HaveOccurred(), "failed to deploy contract")
```

Comment:
> I suggest adding it to the DescribeTables and comparing the call to the ERC-20 vs EVM Extension

@fedekunze - Will deploy the original `ERC20MinterBurnerDecimals` contract here and compare the results 👍 Is that the right one to use?

Reply:
yes